Add study selection to json

peterhambly opened the pull request:
About to merge.

devilgate commented:
This seems like a mammoth set of changes, while the title is quite innocuous. I can't see anything wrong with it, but I'm worried it might conflict badly with my extant changes.

Still, probably best to get it merged so I can find out.

peterhambly commented:
Mainly in the front end to support the saving of the method of study selection via the JSON files. New modal window for entry by postcode/national grid coordinate/WGS84 coordinate (dependent on the geography capabilities) Two new REST calls: *getPostalCodes*, *getPostalCodeCapabilities* for matching by postal code. *RifResultTable.java* gets a new JSON type and a proper JSON escaper. Added support for population weighted centroids to the *getTileMakerCentroids* REST call. Improved asynchronous behaviour in study selection. Add study selection state effectively fixed most of the issues with the old studyState control mechanism (which has deliberately not been modified as it is used to populate the database tables). This may fix issue #52 (it appeared to when I tested it so I added more debug).

devilgate commented:
OK, go ahead and merge.
